src/VisualStudio/TestUtilities/IntegrationService.cs

```diff
@@ -0,0 +1,41 @@
+﻿// Copyright (c) Microsoft.  All Rights Reserved.  Licensed under the Apache License, Version 2.0.  See License.txt in the project root for license information.
+
+using System;
+using System.Collections.Concurrent;
+using System.Reflection;
+using System.Runtime.Remoting;
+
+namespace Roslyn.VisualStudio.Test.Utilities
+{
+    internal class IntegrationService : MarshalByRefObject
```

Comment:
Please add a comment here explaining this type lives in the VS side of things and exists to be the proxy. It makes sense to me since we have chatted about it, but @DustinCampbell can attest to having to debug through other frameworks where there was no documentation on what lived in what process and it drove him insane.